### ISSUE TITLE
Warn if user install multiple maps for the same geographic area

### ISSUE THREAD

**Issue by scaidermern** [osmandapp/OsmAnd#15039]
Having more than one map installed for the same region leads to strange navigation instructions and possibly other problems (see for example [car route: too many WRONG “turn slightly left” every hundred meters](https://community.openstreetmap.org/t/android-osmand-4-1-11-car-route-too-many-wrong-turn-slightly-left-fr-translated-every-hundred-meters/2092?u=scai)).

OsmAnd should present a warning to the use if more than one map for the same location will be installed.

**Comment by sonora** [comment]
Yes, I also observe many rather obscure user issues around the areas of
- OsmAnd Live
- Roads-only maps vs. Full maps
- and regionally overlapping maps since we started to make them available for the sake of offering also smaller downloads.

Not quite clear if we could maybe **_automatically detect and resolve this at least for the purpose of routing_** when a user starts one. But a warning would be a good first step.

**Comment by vshcherb** [comment]
That's of course annoying but also it's a product issue. 
- We try to detect and disable road only maps during rendering if you have normal maps already. 
- If you have 2 consistent maps we remove duplicate roads by ids.
- OsmAnd live is monitored and attached to the regions and automatically disabled if you have newer map

But of course there are unsupported combinations such as different update time for large vs small region.

**Comment by sonora** [comment]
We may want to give that some priority thinking, though. I know of users who for a long time had the impression that OsmAnd's navigation capabilities are completely useless. In many cases, after debugging, it simply boiled down to this issue... See also the forum link @scaidermern included above.

**Comment by scaidermern** [comment]
I fully agree with @sonora. Issues that impact user experience should get a higher priority than fancy new features.

**Comment by vshcherb** [comment]
Hm... But that needs to be explained properly we have warnings before download but it might be an issue that people don't notice warnings and forget later, so it needs to be constant check after start?

**Comment by scaidermern** [comment]
Sounds good. Probably a setting is needed to deactivate this check if people do obscure things with custom maps.

**Comment by dmpr0** [comment]
I propose to remove duplicated maps.
For example, for Belgium, we propose **Standard map** full country map, and additionally **Standard map** for two regions, downloading all standard maps (whole country + regions) will lead to duplicates. 

We already remove whole country maps for some countries due to the size, we need to do the same for all countries divided into regions. 

Additionally we will need to update UI for **Updates** map and **Live updates** screens. Add options to update only one country or select only needed regions.

**Comment by vshcherb** [comment]
Related  to https://github.com/osmandapp/OsmAnd/issues/16299

**Comment by scaidermern** [comment]
This ticket gets moved from version to version without a fix. Fixing it doesn't seem like a big deal to me. This problem is constantly annoying various users. Please fix it finally.

**Comment by vshcherb** [comment]
The fix is on the table - delete combined maps. They shouldn't exist as create too much troubles, so the proper fix would require treating Collection of Germany maps as 1 UI item and refactoring lots of UI.

However simple fix is to delete combined maps leaving only unique - I'm not sure users will appreciate it.

**Comment by sonora** [comment]
Question is how to know about this strategy - how to know which maps to keep and which to remove. Perhaps we  could somehow mark "unsupported/trouble maps" in red color or something.

**Comment by vshcherb** [comment]
well if map won't be updated it's a subject to delete locally :-) Though I don't like when you need to deal with 50 items instaed of 1. However it's already a deal for example Ukraine Indonesia it's subdivided > 20 items and that need to represented in UI as 1 element.

I don't know whether we will have any drawbacks regarding 50+ files - US as 1 item, Obviously the deal comes that partial updates will be an issue, like during update only half of the states were downloaded

**Comment by sonora** [comment]
It's not so much about how big we bundle. I think our issue started when we began to offer overlapping maps, e.g. offered an entire country, but alternatively also separate maps of its regions, and or even of its cities.Users are not keeping track of what they have, so may start with a city, later download the country, and thus create the issue. If all else fails, perhaps we have to go back to strictly offer overlap-free again, finding some compromise in terms of chunk size. But never offer 2 standard maps covering an identical point.

**Comment by scaidermern** [comment]
What I imagined:
* Warn whenever the user tries to download a map that covers a region of another already installed map
* Warn whenever route calculation detects two data sources for the same region (not sure how easy that is)

This probably only needs to be done for routable maps / official maps. Some users may use custom maps that overlap with others but should get ignored during these checks.

**Comment by vshcherb** [comment]
Created separate task https://github.com/osmandapp/OsmAnd/issues/21801 that could be considered as subtask

**Comment by vshcherb** [comment]
We stopped providing regional maps and keep working on UI improvements. On Updates screen regions are groupped by country

**Comment by scaidermern** [comment]
So will there be a warning when installing both standard and roads only map? This happens every now and then, leading to strange routing problems. See https://t.me/OsmAndMaps/55704 for a very recent example.

**Comment by vshcherb** [comment]
Road only maps went out of sight... cause of long history that issue, I think it deserves another issue @Zirochkabila

**Comment by wonczak** [comment]
This case still has relevance, -especially- due to the new split in regional maps.
For example: Germany was divided by states, now some states are subdivided into smaller parts (a necessary move, since the state maps approached 2 GB in size). However: If you update the maps and get the new regional maps, the old state maps remain - leading to the exact problem of now having -two- maps for the same region.
So: The a warning (and maybe a dialog giving the choice to deactivate or delete one of the maps) is in order here.
Same problem for France and Japan. Possibly more countries (I haven't checked).

**Comment by scaidermern** [comment]
> Road only maps went out of sight... cause of long history that issue, I think it deserves another issue [@Zirochkabila](https://github.com/Zirochkabila)

Is there a new ticket for checking if roads-only maps and full maps overlap each other? Or should I create one?

**Comment by scaidermern** [comment]
> Road only maps went out of sight... cause of long history that issue, I think it deserves another issue [@Zirochkabila](https://github.com/Zirochkabila)

@vshcherb @Zirochkabila What is the status here?

**Comment by Zirochkabila** [comment]
@scaidermern This issue is currently being investigated. It may be worth discussing it at https://github.com/osmandapp/OsmAnd/issues/24627
